### PR TITLE
🐛 Fix normalization issue with quoted & case sensitive columns

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
Relates to https://github.com/airbytehq/oncall/issues/84

## How
It seems MSSQL does care about case sensitivity of columns (even quoted ones).
(This might be depending on a "collation" settings of the server too)

But this PR makes it so we always resolve column conflicts by lowercasing all identifiers.

When I added the new test case, it surfaced similar behavior/errors on other destinations, so I'm fixing it there too.
(bigquery/mysql)

## Recommended reading order
1. `x.java`
2. `y.python`

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/postgres/test_simple_streams/first_output/airbyte_incremental/scd/test_normalization/multiple_column_names_conflicts_scd.sql [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      md5(cast(coalesce(cast("id" as 
+    varchar
+), '') as 
+    varchar
+)) as _airbyte_unique_key,
+      "id",
+      "User Id",
+      user_id,
+      "User id",
+      "user id",
```
Comment (tuliren) [inline airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/postgres/test_simple_streams/first_output/airbyte_incremental/scd/test_normalization/multiple_column_names_conflicts_scd.sql:28]: Sorry that I don't understand how the first `User id` is not converted to lower case. The main change in the Python file is this one I think:

```diff
-            if not is_quoted and not self.needs_quotes(input_name):
-                result = input_name.lower()
+            result = input_name.lower()
```

which seems to universally convert names to lower case. Can you elaborate?

**Review by tuliren — APPROVED** [review]

File: airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/postgres/test_simple_streams/first_output/airbyte_incremental/scd/test_normalization/multiple_column_names_conflicts_scd.sql (hunk 1 above)
Comment (ChristopheDuong) [reply airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/postgres/test_simple_streams/first_output/airbyte_incremental/scd/test_normalization/multiple_column_names_conflicts_scd.sql:28]: This file you commented on is the output for Postgres, not mssql.

- In Postgres, when quoted, the identifier's case matters (so  "User Id", "User id" and "user id" are different column names)
- But as the integration test reveals and as the reported On-Call issue reports, MSSQL does not make a difference with these 3 columns names with different cases and considers them all to be equal, and thus conflicting. 

So, to mirror this behavior with MSSQL in normalization code, I universally lowercase all column identifiers for MSSQL even if they are quoted. As a consequence, normalization now detects column name conflicts and will resolve it by renaming conflicting names with extra '_1', '_2' etc

Comment (ChristopheDuong) [reply airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/postgres/test_simple_streams/first_output/airbyte_incremental/scd/test_normalization/multiple_column_names_conflicts_scd.sql:28]: Actually, I'm going to change the PR and isolate it to normalize this sort of casing only when looking for column name conflicts...

File: airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mssql/test_simple_streams/first_output/airbyte_views/test_normalization/multiple_column_names_conflicts_stg.sql [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+-- SQL model to parse JSON blob stored in a single column and extract into separated field columns as described by the JSON Schema
+-- depends_on: "test_normalization".test_normalization._airbyte_raw_multiple_column_names_conflicts
+select
+    json_value(_airbyte_data, ''$."id"'') as id,
+    json_value(_airbyte_data, ''$."User Id"'') as "User Id",
+    json_value(_airbyte_data, ''$."user_id"'') as user_id,
+    json_value(_airbyte_data, ''$."User id"'') as "User id_1",
+    json_value(_airbyte_data, ''$."user id"'') as "user id_2",
+    json_value(_airbyte_data, ''$."UserId"'') as userid,
```
Comment (ChristopheDuong) [inline airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mssql/test_simple_streams/first_output/airbyte_views/test_normalization/multiple_column_names_conflicts_stg.sql:14]: @tuliren here is the output of the new test for MSSQL

File: airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/postgres/test_simple_streams/first_output/airbyte_incremental/scd/test_normalization/multiple_column_names_conflicts_scd.sql (hunk 1 above)
Comment (tuliren) [reply airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/postgres/test_simple_streams/first_output/airbyte_incremental/scd/test_normalization/multiple_column_names_conflicts_scd.sql:28]: I see. Thanks for the explanation.

**Pull request merged**